### PR TITLE
add train demo to paddle_build.sh

### PR DESCRIPTION
**Pull request by mapingshuo**
This pull request adds a training demo of logistic regression task to the continuous integration.

Training with fluid inference lib is an important feature of Paddle, but the [trainer demo](https://github.com/PaddlePaddle/Paddle/tree/develop/paddle/fluid/train/demo) is not included in continuous integration for now.

### CODE REVIEW

File: paddle/fluid/framework/tensor.h [hunk 1]
```diff
@@ -203,6 +181,28 @@ class Tensor {
...
+   * @brief the detail format of memory block which have layout as kMKLDNN
+   *
+   * @note MKLDNN lib support various memory format like nchw, nhwc, nChw8C,
+   *       nChw16c, etc. For a MKLDNN memory block, layout will be set as
+   *       DataLayout::kMKLDNN meanwhile detail memory format will be kept in
+   *       this field.
+   */
+
+  mkldnn::memory::format format_ = mkldnn::memory::format::format_undef;
+#endif
```
Comment (luotao1) [inline paddle/fluid/framework/tensor.h]: why modify the `tensor.h`?

Comment (mapingshuo) [reply paddle/fluid/framework/tensor.h]: There is s segmentation fault with the original tensor.h, as issue discussed: https://github.com/PaddlePaddle/Paddle/issues/19922

Comment (luotao1) [reply paddle/fluid/framework/tensor.h]: @jczaja Could you help review it?

Comment (jczaja) [reply paddle/fluid/framework/tensor.h]: @mapingshuo  This change in tensor.h should not be needed anymore. This was workaround to make demo working , before root cause was found. Root cause was lack of -DPADDLE_WITH_MKLDNN in CMakeLists.txt of demo ci . After this flag was added you should not modify tensor.h . Just adding a flag -DPADDLE_WITH_MKLDNN should be enough (It works on my setup). Please revert tensor.h changes and retest demo_trainer .

**Review by jczaja — CHANGES_REQUESTED** [review]

File: paddle/fluid/framework/tensor.h (hunk 1 above)
Comment (mapingshuo) [reply paddle/fluid/framework/tensor.h]: Thank you, I understand now!

**Review by luotao1 — APPROVED** [review]
LGTM

File: paddle/fluid/train/demo/run.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+PADDLE_ROOT=$1
+TURN_ON_MKL=$2 # use MKL or Openblas
+
+# download models
+function download() {
+    wget -q http://paddle-tar.bj.bcebos.com/train_demo/LR/main_program
+    wget -q http://paddle-tar.bj.bcebos.com/train_demo/LR/startup_program
+}
+
+download
```
Comment (luotao1) [inline paddle/fluid/train/demo/run.sh:14]: download下一个PR可以考虑做缓存，虽然这部分不多。

Comment (mapingshuo) [reply paddle/fluid/train/demo/run.sh:14]: 多谢！

File: paddle/fluid/framework/tensor.h (hunk 1 above)
Comment (mapingshuo) [reply paddle/fluid/framework/tensor.h]: > @mapingshuo This change in tensor.h should not be needed anymore. This was workaround to make demo working , before root cause was found. Root cause was lack of -DPADDLE_WITH_MKLDNN in CMakeLists.txt of demo ci . After this flag was added you should not modify tensor.h . Just adding a flag -DPADDLE_WITH_MKLDNN should be enough (It works on my setup). Please revert tensor.h changes and retest demo_trainer .

I have restored tensor.h, please review this PR again.

Comment (mapingshuo) [reply paddle/fluid/framework/tensor.h]: @jczaja

**Review by guru4elephant — APPROVED** [review]

**Review by jczaja — APPROVED** [review]
LGTM

**Pull request merged**